torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -50,23 +50,37 @@ std::shared_ptr<FutureMessage> RequestCallbackImpl::processRpc(
 
       // scriptCall is only alive within this block, use reference to avoid copy
       auto& stack = scriptCall.stackRef();
+
       if (scriptCall.hasOp()) {
         scriptCall.op()->getOperation()(stack);
-      } else {
-        PythonRpcHandler::getInstance()
-            .jitCompilationUnit()
-            ->get_function(scriptCall.qualifiedName())
-            .run(stack);
+        TORCH_INTERNAL_ASSERT(
+            stack.size() == 1,
+            "Return value of a builtin operator or a "
+            "TorchScript function should be a single IValue, got a vector of "
+            "size ",
+            stack.size());
+        return wrap(
+            std::move(ScriptResp(std::move(stack.front()))).toMessage());
       }
 
-      TORCH_INTERNAL_ASSERT(
-          stack.size() == 1,
-          "Return value of a builtin operator or a "
-          "TorchScript function should be a single IValue, got a vector of "
-          "size ",
-          stack.size());
-
-      return wrap(std::move(ScriptResp(std::move(stack.front()))).toMessage());
+      auto future = PythonRpcHandler::getInstance()
+                        .jitCompilationUnit()
+                        ->get_function(scriptCall.qualifiedName())
+                        .runAsync(stack);
```

Comment:
Do we need to worry about the life time of the stack? Will it be possible that the stack is used after the deletion of `scriptCall`?

Reply:
I believe we're fine, please in particular see 
torch/csrc/jit/interpreter.cpp:1143, where we end up effectively move-copying the stack over into the continuation.